common/web/gesture-recognizer/src/gestureRecognizerConfiguration.ts

```diff
@@ -0,0 +1,23 @@
+namespace com.keyman.osk {
+  export interface GestureRecognizerConfiguration {
+    /**
+     * Specifies the element that mouse input listeners should be attached to.  If
+     * not specified, `eventRoot` will be set equal to `targetRoot`.
+     */
+    readonly mouseEventRoot?: HTMLElement;
+
+    /**
+     * Specifies the element that touch input listeners should be attached to.  If
+     * not specified, `eventRoot` will be set equal to `targetRoot`.
+     */
+    readonly touchEventRoot?: HTMLElement;
+
+    /**
+     * Specifies the most specific common ancestor element of any event target
+     * that the `InputEventEngine` should consider.
+     */
+    readonly targetRoot: HTMLElement;
+
+    readonly coordConstrainedWithinInteractiveBounds: (coord: InputEventCoordinate) => boolean;
```

Comment:
Can we add a comment for this as well, given I don't really know what it means?

Reply:
We _could_... but it's going away in the next PR.

Given some of your comments on that "next PR", though... maybe I should write a minor doc about how that operates.  KMW's OSK has long had some pretty involved logic for determining when to auto-cancel ongoing touch events, and the next PR is _all about_ providing a clean, clearer abstraction of that.  If you're impatient, check out the design of #6842's host page and its user tests; the visual design of the interactive area aims to make the abstraction more visually-accessible.

Reply:
I've added a new documentation file in #6844 that should help elucidate the design it lands upon, assuming the new comments there aren't already enough in that regard.